src/test/scala/chiselTests/PrintableSpec.scala

```diff
@@ -47,88 +49,93 @@ class PrintableSpec extends AnyFlatSpec with Matchers {
         case _ => fail(s"Regex to process Printf should work on $str!")
       }
     }
-
     firrtl.split("\n").collect {
       case PrintfRegex(matched) =>
         val (str, args) = processBody(matched)
         Printf(str, args)
     }
   }
 
+
+  // Generates firrtl, gets Printfs
+  // Returns None if failed match; else calls the partial function which could have its own check
+  // Returns Some(true) to caller 
+  // Not calling fail() here - and letting caller do so - helps in localizing errors correctly. 
+  private def generateAndCheck(gen : => RawModule)(check : PartialFunction[Seq[Printf],Unit]) = {
+    val firrtl = ChiselStage.emitChirrtl(gen)
+    val printfs = getPrintfs(firrtl)
+    if(!check.isDefinedAt(printfs)) {
+      println("Firrtl = \n",firrtl)
+      println("Actual = ",printfs)
```

Comment:
Remove these before merging

Reply:
These messages are useful when failures occur even in the future - let me try to use Scala's Either API return this along with error object (Left in this case).

Reply:
Actually it might get too complicated - for now will just remove these before merging and in a future PR enhance the test framework!